package/config_types/config_object.js

```diff
@@ -0,0 +1,49 @@
+const objectify = require('../utils/objectify')
+const deepAssign = require('../utils/deep_assign')
+const deepMerge = require('../utils/deep_merge')
+const { isStrPath, prettyPrint } = require('../utils/helpers')
+
+/**
+  * @class
+  * @extends { Object }
+*/
+class ConfigObject extends Object {
```

Comment:
Extending native `Object` (and `Array` above) could have unexpected consequences. I don't have any references to point to, but it's generally not something I do. Do you have any thoughts?  Alternatively, these could be bare classes that manage an object internally.

Reply:
Yeah, I read a bit about it but it seems for our use case it's safe to use since the usage is limited only to webpack config and unlike `prototype` it doesn't change native objects. 

I tried using bare classes with internal object but ran into some gotchas. We basically need native types with some custom methods on it. 

Do you think it's bad for our use case?